### PR TITLE
Initial GKE setup scripts

### PR DESCRIPTION
**Pull request by cromwellian**
Allows anyone to spin-up an arcs cloud cluster on GCP.

---
This change is [Reviewable](https://reviewable.io/reviews/polymerlabs/arcs/2156)

### CODE REVIEW

**Comment by cromwellian** [comment]
These scripts automate as much as possible:

1) automatic syncing of dynamically assigned IP addresses from GCP and external domain names (external-dns)
2) automatic provisioning an nginx load balancer with external address (nginx-ingress)
3) automatic acquisition of letsencrypt TLS certs for this external domain/ip address (cert-manager)
4) Building and pushing docker images to GCP repo
5) Spinning up a deployment of arcs-master
6) One-time setup scripts to create the GCP cluster and service accounts  + DNS zones

File: server/deployment/setup_gke.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    --max-nodes 10 \
+    --enable-autoscaling \
+    --enable-autorepair \
+    --enable-autoupgrade \
+    --enable-cloud-logging \
+    --enable-cloud-monitoring \
+    --service-account=$SERVICE_ACCOUNT
+
+# Download credentials for kubernetes
+echo "Fetching Kubernetes credentials for clusrter..."
```
Comment (shaper) [inline server/deployment/setup_gke.sh]: cluster (typo)

File: server/deployment/setup_helm.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+source ./environment.sh
+
+source ./configure_helm_permissions.sh
+
+# Initialize helm
+echo "Initialing Helm"
```
Comment (shaper) [inline server/deployment/setup_helm.sh]: Initializing (typo)

File: server/deployment/build.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
+source ./environment.sh
+
+cd ../..
+echo "Building Arcs Master Docker Image..."
+docker build . -t $BUILD_LABEL
```
Comment (shaper) [inline server/deployment/build.sh]: Check return codes here and below? Similar rest of scripts (could do in follow-on PR)

File: server/deployment/README.md [hunk 4]
```diff
@@ -1,7 +1,47 @@
...
+
+Edit environment.sh to configure your your service account, dns zone, project id, etc.
+
+== Run setup scripts once ==
+
+Finally run setup_gke.sh (can only run once)
+
+== Edit/Refresh ==
+
+Runing build.sh builds and pushes docker images to GCP. The services should notice when
```
Comment (shaper) [inline server/deployment/README.md]: Running (typo)

File: server/deployment/README.md [hunk 5]
```diff
@@ -1,7 +1,47 @@
...
+
+== Configuration Prerequisites ==
+
+Create a GCP account and enable billing. 
+
+Then run 'gcloud init'
+Then run 'gcloud components install kubectl'
+Then run 'gcloud components update'
+
+Create a domain on Google Domains registrar. Next, Setup a GCP CloudDNS zone for your
```
Comment (shaper) [inline server/deployment/README.md]: Perhaps we should note that currently the configs are hard coded (?) for `skarabrae.org` as just an initial MVP, and that files X, Y, Z, etc. would need to be updated to reflect the alternate domain?

Could also be worth filing an issue to genericize the domain bits and referencing it here.

**Pull request merged**